controllers/datasciencecluster/datasciencecluster_controller.go

```diff
@@ -635,6 +654,22 @@ func (r *DataScienceClusterReconciler) watchDefaultIngressSecret(ctx context.Con
 	return nil
 }
 
+func (r *DataScienceClusterReconciler) watchIngressResources(ctx context.Context, a client.Object) []reconcile.Request {
+	requestName, err := r.getRequestName(ctx)
```

Comment:
unrelated, but `getRequestName` seems incorrect as it would return a result even if there's no actual instances of a DSC

Reply:
let me make another PR to follow this 0 instance case. 
the original code was added 10 months ago.....

Reply:
follow up in https://github.com/opendatahub-io/opendatahub-operator/pull/1323